File: frontend/lib/domain/models/diary/note_element.dart
import 'dart:io';

/// Classe base astratta per gli elementi delle note.
abstract class NoteElement {
  /// Contenuto dell'elemento (testo, path dell'immagine o dell'audio).
  /// Essendo pubblica, Dart crea in automatico getter e setter efficienti.
  String content;
  File? file;
  String? noteParentId;
  String? noteElementId;

  NoteElement(this.content,  {this.file, this.noteParentId, this.noteElementId});

  /// Getter astratto che definisce il tipo dell'elemento.
  /// Ogni sottoclasse DEVE implementarlo.
  String get type;

}

File: backend/src/diary/domain/note_element.py
from .diary_type import DiaryType

class NoteElement:
    def __init__(
        self,
        note_id: str,
        note_element_id: str,
        type: str,
        content: str,
    ):
        self.note_id = note_id
        self.note_element_id = note_element_id
        self.type = type
        self.content = content